server/persistence/iwclient/iwclient_list.go

```diff
@@ -36,26 +38,33 @@ func (c *Client) fetchMissingWorkspaces(ctx context.Context, user string, worksp
 		return err
 	}
 
+	// fetch all workspaces
+	aww := workspacesv1alpha1.InternalWorkspaceList{}
+	if err := c.backend.List(ctx, &aww); err != nil {
+		return err
+	}
+
 	// filter already fetched Workspaces
-	fsbb := make([]*toolchainv1alpha1.SpaceBinding, 0, len(sbb.Items))
+	fsp := make([]string, 0, len(sbb.Items))
 	for i, sb := range sbb.Items {
 		if slices.ContainsFunc(workspaces.Items, func(w workspacesv1alpha1.InternalWorkspace) bool {
-			return w.Name == sb.Spec.Space
+			return w.Status.Space.Name == sb.Spec.Space
 		}) {
 			continue
 		}
 
-		fsbb = append(fsbb, &sbb.Items[i])
+		fsp = append(fsp, sbb.Items[i].Spec.Space)
 	}
+	sort.Strings(fsp)
+	fsp = slices.CompactFunc(fsp, func(s1, s2 string) bool { return cmp.Compare(s1, s2) <= 0 })
```

Comment:
Why are we sorting?  I don't think list requests have any requirement about the sort order of the returned resources.

Reply:
that's just because of how [CompactFunc|https://pkg.go.dev/golang.org/x/exp/slices#CompactFunc] and [Compact|https://pkg.go.dev/golang.org/x/exp/slices#Compact] work.

> Compact replaces consecutive runs of equal elements with a single copy. This is like the uniq command found on Unix.

Reply:
also it should impact the order of returned items

Reply:
as discussed, let's try simplify this code using a Set

Reply:
done in a2d68ed82f70da314a2c6aba0fb9ecc8c5c95bb0